recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,300 @@
+            self._cmake.definitions["UA_ENABLE_ENCRYPTION_MBEDTLS"] = True
+
+
+        self._cmake.definitions["UA_MULTITHREADING"] = self._ua_multithreaded()
+
+        self._cmake.definitions["UA_ENABLE_AMALGAMATION"] = self.options.enable_amalgamation
+        self._cmake.definitions["UA_ENABLE_HISTORIZING"] = self.options.enable_historizing
+        self._cmake.definitions["UA_ENABLE_METHODCALLS"] = self.options.enable_methodcalls
+        self._cmake.definitions["UA_ENABLE_NODEMANAGEMENT"] = self.options.enable_nodemanagement
+        self._cmake.definitions["UA_ENABLE_SUBSCRIPTIONS"] = self.options.enable_subscriptions
+        self._cmake.definitions["UA_ENABLE_SUBSCRIPTIONS_EVENTS"] = self.options.enable_subscriptions_events
+        self._cmake.definitions["UA_ENABLE_DISCOVERY"] = self.options.enable_discovery
+        self._cmake.definitions["UA_ENABLE_DISCOVERY_MULTICAST"] = self.options.enable_discovery_multicast
+        self._cmake.definitions["UA_ENABLE_MICRO_EMB_DEV_PROFILE"] = self.options.enable_micro_emb_dev_profile
+        self._cmake.definitions["UA_ENABLE_QUERY"] = self.options.enable_query
+        self._cmake.definitions["UA_ENABLE_IMMUTABLE_NODES"] = self.options.enable_immutable_nodes
+        self._cmake.definitions["UA_ENABLE_EXPERIMENTAL_HISTORIZING"] = self.options.enable_experimental_historizing
+        self._cmake.definitions["UA_ENABLE_PUBSUB"] = self.options.enable_pubsub
+        self._cmake.definitions["UA_ENABLE_PUBSUB_ETH_UADP"] = self.options.enable_pubsub_eth_uadp
+        self._cmake.definitions["UA_ENABLE_PUBSUB_DELTAFRAMES"] = self.options.enable_pubsub_deltaframes
+        self._cmake.definitions["UA_ENABLE_PUBSUB_INFORMATIONMODEL"] = self.options.enable_pubsub_informationmodel
+        self._cmake.definitions["UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS"] = self.options.enable_pubsub_informationmodel_methods
+        self._cmake.definitions["UA_ENABLE_PUBSUB_CUSTOM_PUBLISH_HANDLING"] = self.options.enable_pubsub_custom_publish_handling
+        self._cmake.definitions["UA_ENABLE_JSON_ENCODING"] = self.options.enable_json_encoding
+        self._cmake.definitions["UA_ENABLE_STATUSCODE_DESCRIPTIONS"] = self.options.enable_statuscode_descriptions
+        self._cmake.definitions["UA_ENABLE_NODESET_COMPILER_DESCRIPTIONS"] = self.options.enable_nodeset_compiler_descriptions
+        self._cmake.definitions["UA_ENABLE_MALLOC_SINGLETON"] = self.options.enable_malloc_singleton
+        self._cmake.definitions["UA_MSVC_FORCE_STATIC_CRT"] = self.options.msvc_force_static_crt
+        self._cmake.definitions["UA_ENABLE_DISCOVERY_SEMAPHORE"] = self.options.enable_discovery_semaphore
+        self._cmake.definitions["UA_DEBUG_DUMP_PKGS"] = self.options.debug_dump_pkgs
+        self._cmake.definitions["UA_ENABLE_HARDENING"] = self.options.enable_hardening
+        self._cmake.definitions["OPEN62541_VERSION"] = self.version
+        self._cmake.definitions["UA_DEBUG" ] = self.settings.build_type == "Debug"
```

Comment:
Shouldn't this be set to Release? Are you running the internal test and thus need this flag (you shouldn't run unit tests on conan though...). 

Also if I am not mixing things up, isn't this flag also changing optimization levels? 

If you really want to allow the user to change from release to debug, set it through `settings.build_type` instead

Reply:
Also, this option is set by [CMakeLists.txt](https://github.com/open62541/open62541/blob/master/CMakeLists.txt#L387) file itself when `CMAKE_BUILD_TYPE` is Debug, so you can skip this option, since it is already handled by `settings.build_type`